instrumentation/aws-java-sdk-sqs-2.1.0/src/main/java/software/amazon/awssdk/services/sqs/DefaultSqsAsyncClient_Instrumentation.java

```diff
@@ -74,8 +103,195 @@ public CompletableFuture<ReceiveMessageResponse> receiveMessage(ReceiveMessageRe
         return result.whenComplete(new BiConsumer<ReceiveMessageResponse, Throwable>() {
             @Override
             public void accept(ReceiveMessageResponse receiveMessageResponse, Throwable throwable) {
-                MetricUtil.finishSegment(segment);
+                SqsV2Util.finishSegment(segment);
             }
         });
     }
+
+    /*
+    * Headers are used as inner classes to avoid class loading errors when running instrumentation tests
+    * */
+
+    public class SQSRequestHeaders implements Headers {
+
+        private SendMessageRequest request = null;
+        private Map<String, String> additionalAttributes = new Hashtable<String, String>();
+
+        public SQSRequestHeaders(SendMessageRequest req) {
+            request = req;
+        }
+
+        @Override
+        public HeaderType getHeaderType() {
+            return HeaderType.MESSAGE;
+        }
+
+        @Override
+        public String getHeader(String name) {
+            Map<String, MessageAttributeValue> messageAttributes = request.messageAttributes();
+            if(messageAttributes != null) {
+                MessageAttributeValue value = messageAttributes.get(name);
+                if (value != null && value.dataType().equalsIgnoreCase("string")) {
+                    String valueString = value.stringValue();
+                    if (valueString != null) {
```

Comment:
This if check is redundant. If valueString is null, the method will return null anyway.

Reply:
Removed the null check